sos/plugins/openvswitch.py

```diff
@@ -104,32 +131,93 @@ def setup(self):
+                "fp-cli conf compiled",
+                "fp-cli iface",
+                "ovs-appctl memory/show",
+            ])
+            self.add_journal(units="virtual-accelerator")
+            for table in ['filter', 'mangle', 'raw', 'nat']:
+                self.add_cmd_output(["fpcmd nf4-rules %s" % table])
+
+            # 6wind doesn't care on which bridge the ports are, there's only
+            # one bridge and it's alubr0
+            port_list = self.call_ext_prog("fp-cli fp-vswitch-ports")
+            if port_list['status'] == 0:
+                for port in port_list['output'].splitlines():
+                    m = re.match(r'^([\d]+):[\s]+([^\s]+)', port)
+                    if m:
+                        port_name = m.group(2)
+                        self.add_cmd_output([
+                            "fp-cli dpdk-cp-filter-budget %s" % port_name,
+                        ])
+
+        # Gather the datapath information for each datapath
+        dp_list_result = self.call_ext_prog('ovs-appctl dpctl/dump-dps')
+        if dp_list_result['status'] == 0:
+            for dp in dp_list_result['output'].splitlines():
+                self.add_cmd_output([
+                    "%s dpctl/show -s %s" % (actl, dp),
+                    "%s dpctl/dump-flows -m %s" % (actl, dp),
+                    "%s dpctl/dump-conntrack -m %s" % (actl, dp),
+                    "%s dpctl/ct-stats-show -m %s" % (actl, dp),
+                    "%s dpctl/ipf-get-status %s" % (actl, dp),
+                ])
+
         # Gather additional output for each OVS bridge on the host.
-        br_list_result = self.collect_cmd_output("ovs-vsctl list-br")
+        br_list_result = self.call_ext_prog("ovs-vsctl -t 5 list-br")
         if br_list_result['status'] == 0:
             for br in br_list_result['output'].splitlines():
                 self.add_cmd_output([
-                    "ovs-appctl fdb/show %s" % br,
+                    "%s bridge/dump-flows --offload-stats %s" % (actl, br),
+                    "%s dpif/show-dp-features %s" % (actl, br),
+                    "%s fdb/show %s" % (actl, br),
+                    "%s fdb/stats-show %s" % (actl, br),
+                    "%s mdb/show %s" % (actl, br),
                     "ovs-ofctl dump-flows %s" % br,
                     "ovs-ofctl dump-ports-desc %s" % br,
                     "ovs-ofctl dump-ports %s" % br,
                     "ovs-ofctl queue-get-config %s" % br,
                     "ovs-ofctl queue-stats %s" % br,
                     "ovs-ofctl show %s" % br,
-                    "ovs-appctl fdb/stats-show %s" % br
                 ])
 
                 # Flow protocols currently supported
                 flow_versions = [
                     "OpenFlow10",
                     "OpenFlow11",
                     "OpenFlow12",
-                    "OpenFlow13"
+                    "OpenFlow13",
+                    "OpenFlow14",
+                    "OpenFlow15"
```

Comment:
Is this intentional, or a result of copying the `master` plugin over the legacy one?

Reply:
This was semi-intentional, I don't see any downside in adding this.